File: package/java/src/main/java/com/programknock/SushiSeating.java
package com.programknock;

import java.util.*;

public class SushiSeating {

    public static List<String> sushiSeating(List<String> commands) {
        Queue<String> waitingQueue = new LinkedList<>(); // FIFO queue for waiting customers
        List<String> seatedCustomers = new ArrayList<>(); // List of seated customers in order

        for (String command : commands) {
            String[] parts = command.split(" ", 2);

            if (parts[0].equals("arrive")) {
                String name = parts[1];
                // Add to queue only if not already in queue
                if (!waitingQueue.contains(name)) {
                    waitingQueue.add(name);
                }
            } else if (parts[0].equals("seat")) {
                int n = Integer.parseInt(parts[1]);
                // Seat up to n customers from the front of the queue
                int seatsToFill = Math.min(n, waitingQueue.size());
                for (int i = 0; i < seatsToFill; i++) {
                    String customer = waitingQueue.poll();
                    if (customer != null) {
                        seatedCustomers.add(customer);
                    }
                }
            }
        }

        return seatedCustomers;
    }
}

File: package/typescript/src/sushiSeating.ts
/**
 * 回転寿司屋の席案内システムを実装する関数。
 *
 * @param commands - 到着・席案内のコマンドリスト
 * @returns 案内されたお客さんのリスト
 */
export function sushiSeating(commands: string[]): string[] {
  const waitingQueue: string[] = []; // FIFO queue for waiting customers
  const seatedCustomers: string[] = []; // List of seated customers in order

  for (const command of commands) {
    const parts = command.split(' ');

    if (parts[0] === 'arrive') {
      const name = parts[1];
      // Add to queue only if not already in queue
      if (!waitingQueue.includes(name)) {
        waitingQueue.push(name);
      }
    } else if (parts[0] === 'seat') {
      const n = parseInt(parts[1], 10);
      // Seat up to n customers from the front of the queue
      const customersToSeat = Math.min(n, waitingQueue.length);
      for (let i = 0; i < customersToSeat; i++) {
        const customer = waitingQueue.shift()!;
        seatedCustomers.push(customer);
      }
    }
  }

  return seatedCustomers;
}
